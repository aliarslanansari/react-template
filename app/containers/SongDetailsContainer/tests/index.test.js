/**
 *
 * Tests for SongDetailsContainer
 *
 */
import React from 'react';
import { timeout, renderProvider } from '@utils/testUtils';
import { SongDetailsContainerTest as SongDetailsContainer } from '../index';

describe('<SongDetailsContainer /> tests', () => {
  let submitSpy;

  beforeEach(() => {
    submitSpy = jest.fn();
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(
      <SongDetailsContainer match={{ params: { id: 123 } }} dispatchGetSong={submitSpy} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should dispatch dispatchGetSong when component renders', async () => {
    renderProvider(<SongDetailsContainer match={{ params: { id: 123 } }} dispatchGetSong={submitSpy} />);
    await timeout(500);
    expect(submitSpy).toBeCalled();
  });
});
