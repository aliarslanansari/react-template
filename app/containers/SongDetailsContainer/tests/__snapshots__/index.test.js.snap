// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SongDetailsContainer /> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div>
      <div
        class="SongDetailsContainer__Container-sc-1gnl6st-0 bLbcEs"
      >
        <div
          class="ant-skeleton ant-skeleton-active"
        >
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 38%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li
                style="width: 61%;"
              />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
